src/Drum.java

```diff
@@ -0,0 +1,36 @@
+import java.util.ArrayList;
+import java.util.Random;
+
+public class Drum {
+
+    private ArrayList<Ball> allBalls = new ArrayList<>();
+    private Random generator = new Random();
```

Comment:
```suggestion
    private final List<Ball> allBalls = new ArrayList<>();
    private final Random generator = new Random();
```

Reply:
[Deze link](https://stackoverflow.com/questions/3383726/java-declaring-from-interface-type-instead-of-class) legt t wel aardig uit